### PR TITLE
Add test cases

### PR DESCRIPTION
**Pull request by lujiajing1126**
In this pull request, I add some test cases to test input and output.
For testability, I use `rxjs.Subject` to handle all the output
and use `setter` to emit them which makes output test easy.

Concerning the support for `node-v6`, `node-v8`, `node-v10`, I just employ
`core-js` polyfill to support some upcoming features of `Array` and `Object` instead
of introducing `Babel` which is indeed too "heavy".

For the test coverage, pls see https://codecov.io/gh/lujiajing1126/redis-cli/branch/test-cases

### CODE REVIEW

File: lib/redis.js [hunk 1]
```diff
@@ -54,73 +93,74 @@ class RedisClient {
...
-			} else {
-				rl.setPrompt(`${this._host}> `);
+		this.rl.on('line', (line) => {
+			this._handleInput(line);
+		}).on('close', () => {
+			// trigger when `SIGINT` received
+			this.next = '\nAbort!';
+			this.next = __NORMAL_EXIT__;
+		});
+		return Promise.resolve(this.rl);
```
[the remark this answers is not in the document]
Comment (lujiajing1126) [reply lib/redis.js]: Yes. You are right.

File: package.json [hunk 2]
```diff
@@ -5,7 +5,7 @@
   "main": "index.js",
   "scripts": {
     "start": "node index.js",
-    "test": "jest"
+    "test": "jest --ci --testResultsProcessor=\"jest-junit\""
```
[the remark this answers is not in the document]
Comment (lujiajing1126) [reply package.json]: I see!

**Comment by lujiajing1126** [comment]
> I also got this error once we I ran the tests:
> 
> ```
> ● readline tests › Test Empty Command
> 
>     expect(jest.fn()).toHaveBeenCalled()
> 
>     Expected mock function to have been called, but it was not called.
> 
>       154 |     it('Test Empty Command', () => {
>       155 |         redisClient._handleInput("");
>     > 156 |         expect(spy.next).toHaveBeenCalled();
>           |                          ^
>       157 |         expect(spy.next.mock.calls[0][0]).toBe(__PR__);
>       158 |     })
>       159 | });
> 
>       at Object.toHaveBeenCalled (test/redis.test.js:156:26)
> ```

Can you try again with the latest commit?
Maybe due to the `break` problem.

File: lib/redis.js [hunk 3]
```diff
@@ -54,73 +93,74 @@ class RedisClient {
...
-					}
-				} catch (err) {
-					console.log(colors.red(`(error) ${err.message}`));
-					rl.prompt();
+			let commands = splitargs(command);
+			if (commands.length !== 0) { // we have commands, so process, otherwise just a new prompt
+				let CMD = commands.shift().toLowerCase();
+				//`exit` and `clear` are not true commands, just part of REPL
+				if (CMD === 'exit') {
+					this._redis_client.quit();
```
[the remark this answers is not in the document]
Comment (gkorland) [reply lib/redis.js]: ok

[the remark this answers is not in the document]
Comment (gkorland) [reply lib/redis.js]: @lujiajing1126 perhaps you should add a comment in this line

[the remark this answers is not in the document]
Comment (lujiajing1126) [reply lib/redis.js]: Added

**Pull request merged**